tests/lava/proc/plateau/test_models.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+
+import unittest
+import numpy as np
+from numpy.testing import assert_almost_equal
+
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+from lava.magma.core.model.py.ports import PyOutPort, PyInPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.process.ports.ports import OutPort, InPort
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.variable import Var
+from lava.magma.core.resources import CPU
+from lava.proc.plateau.process import Plateau
+from lava.proc.dense.process import Dense
+from lava.magma.core.run_configs import Loihi2SimCfg, RunConfig
+from lava.magma.core.run_conditions import RunSteps
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.tests.lava.proc.lif.test_models import VecSendProcess, VecRecvProcess
+
+
+class SpikeGen(AbstractProcess):
```

Comment:
Could you check whether the Spiker Process in
'lava\src\lava\proc\spiker\process.py'
already implements the behavior that you expect from SpikeGen?

Reply:
Since these two processes are very similar, it would be good to avoid redundant processes.

Reply:
Maybe better to extend the Spiker, inherit from it, or write a common abstract spiker, rather than having two independent processes.

Reply:
I've replaced the custom SpikeGen process with implementing source RingBuffers. I didn't use the Spiker because that sends periodic spikes and I only wanted single spikes to be sent.